test/celeritas/CMakeLists.txt

```diff
@@ -428,34 +428,72 @@ celeritas_add_test(user/StepCollector.test.cc
 #-----------------------------------------------------------------------------#
 
 set(_root_data)
+set(_force_update_root_data)
 if(CELERITAS_USE_Geant4 AND CELERITAS_USE_ROOT)
   foreach(_filename "four-steel-slabs" "simple-cms" "lar-sphere")
     set(_gdml "${PROJECT_SOURCE_DIR}/test/geocel/data/${_filename}.gdml")
     set(_basename "${CMAKE_CURRENT_SOURCE_DIR}/data/${_filename}")
     set(_output "${_basename}.root")
-    add_custom_command(
-      OUTPUT "${_output}"
-      DEPENDS
-        celer-export-geant
-        "${_gdml}"
-        "${_basename}.geant.json"
-        "${PROJECT_SOURCE_DIR}/src/celeritas/ext/RootInterfaceLinkDef.h"
-      COMMAND
-        "${CMAKE_COMMAND}" "-E" "env" ${CELER_G4ENV}
-        "$<TARGET_FILE:celer-export-geant>"
-        "${_gdml}"
-        "${_basename}.geant.json"
-        "${_output}"
-        "--gen-test"
+    set(_target_name "root-test-data-${_filename}")
+
+    set(_update_script
+      "${CMAKE_CURRENT_BINARY_DIR}/update-${_target_name}-$<CONFIG>.cmake")
+    set(_update_deps
+      "${_update_script}" 
+      celer-export-geant
+      "${_gdml}"
+      "${_basename}.geant.json"
+      "${PROJECT_SOURCE_DIR}/src/celeritas/ext/RootInterfaceLinkDef.h"
+    )
+
+    # Defer update call to a script so that root files aren't targeted by
+    # cmake-clean
+
+    file(GENERATE
+      OUTPUT "${_update_script}"
+      CONTENT
+        "if(NOT EXISTS \"${_output}\")
+           execute_process(
+             COMMAND
+               \"${CMAKE_COMMAND}\" \"-E\" \"env\" ${CELER_G4ENV}
+               \"$<TARGET_FILE:celer-export-geant>\"
+               \"${_gdml}\"
+               \"${_basename}.geant.json\"
+               \"${_output}\"
+               \"--gen-test\"
+         )
+         endif()"
+    )
+
+    add_custom_target("${_target_name}"
+      DEPENDS ${_update_deps}
+      COMMAND "${CMAKE_COMMAND}" "-P" "${_update_script}"
+      VERBATIM
+    )
+
+    list(APPEND _root_data "${_target_name}")
+
+    # Can't order targets in CMake: use commands + script instead
+
+    add_custom_target("force-update-${_target_name}"
+      DEPENDS ${_update_deps}
+      COMMAND "${CMAKE_COMMAND}" "-E" "rm" "${_output}"
+      COMMAND "${CMAKE_COMMAND}" "-P" "${_update_script}"
       VERBATIM
     )
-    list(APPEND _root_data "${_output}")
+
+    list(APPEND _force_update_root_data "force-update-${_target_name}")
   endforeach()
 endif()
 
 add_custom_target(update-root-test-data
-  COMMENT "Regenerate ROOT test data"
+  COMMENT "Regenerate ROOT test data if it doesn't exist"
   DEPENDS ${_root_data}
 )
 
+add_custom_target(force-update-root-test-data
+  COMMENT "Force regeneration of ROOT test data"
```

Comment:
This was the intent of `update-root-test-data`; to forcibly regenerate it on command. It's also what we need to call from the CI workflow to test that the ROOT I/O is in use.

Reply:
If we only ever want to overwrite, that simplifies the logic I think.